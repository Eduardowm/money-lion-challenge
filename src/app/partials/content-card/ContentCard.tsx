import { ContentCard as ContentCardType } from '@/data/entities/ContentCardEntity'
import ContentCardComments from '../content-card-comments/ContentCardComments'
import ContentCardContent from '../content-card-content/ContentCardContent'
import ContentCardImage from '../content-card-image/ContentCardImage'

interface ContentCardProps {
  post: ContentCardType
}

function ContentCard({ post }: ContentCardProps) {
  return (
    <div className="flex flex-col gap-2 p-2 border border-gray-800 rounded">
      <ContentCardImage source={post.image} alt={post.title} />
      <ContentCardContent post={post} />

      <hr className="border-gray-800 mx-[-8px]" />

      <ContentCardComments comments={post.comments} />
    </div>
  )
}
export default ContentCard
