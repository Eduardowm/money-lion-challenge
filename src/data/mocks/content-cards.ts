import { ContentCard } from '../entities/ContentCardEntity'

export const mockRawContentCard = {
  id: 'abc-def-ghi',
  imageUri: '/images/mock.jpg',
  textData: {
    title: 'A mock title',
    subTitle: 'A mock subtitle',
    body: 'A mock content with a lot of text to fill the space',
    author: { first: 'Mock', last: 'author' },
  },
  metadata: {
    priority: 1,
    publishDate: '2021-01-01',
  },
  comments: [
    {
      text: 'A mock comment',
      author: 'Commenter 1',
      profilePic: '/images/mock.jpg',
      likes: 20,
    },
    {
      text: 'The second mock comment',
      author: 'Commenter 2',
      profilePic: '/images/mock2.jpg',
      likes: 50,
    },
  ],
}

export const mockContentCards: ContentCard[] = [
  {
    id: 'abc-def-ghi',
    title: 'A mock title',
    subtitle: 'A mock subtitle',
    content: 'A mock content with a lot of text to fill the space',
    priority: 1,
    author: {
      name: 'Mock author',
    },
    image: '/images/mock.jpg',
    comments: [
      {
        name: 'Commenter 1',
        image: '/images/mock.jpg',
        text: 'A mock comment',
        likes: 20,
      },
      {
        name: 'Commenter 2',
        image: '/images/mock2.jpg',
        text: 'The second mock comment',
        likes: 50,
      },
    ],
  },
  {
    id: 'ihg-fed-cba',
    title: 'An alternative mock title',
    subtitle: 'An alternative mock subtitle',
    content: 'An alternative mock content with a lot of text to fill the space',
    priority: 1,
    author: {
      name: 'An alternative mock author',
    },
    image: '/images/mock.jpg',
    comments: [
      {
        name: 'Alternative commenter 1',
        image: '/images/mock.jpg',
        text: 'An alternative mock comment',
        likes: 20,
      },
    ],
  },
]
