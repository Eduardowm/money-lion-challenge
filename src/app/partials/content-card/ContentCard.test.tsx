import { render } from '@testing-library/react'

import { mockContentCards } from '@/data/mocks/content-cards'
import ContentCard from './ContentCard'

describe('ContentCard', () => {
  it('renders as expected', () => {
    const { container } = render(<ContentCard post={mockContentCards[0]} />)

    expect(container).toMatchSnapshot()
  })
})
