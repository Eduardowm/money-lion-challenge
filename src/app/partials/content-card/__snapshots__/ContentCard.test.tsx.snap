// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentCard renders as expected 1`] = `
<div>
  <div
    class="flex flex-col gap-2 p-2 border border-gray-800 rounded"
  >
    <div
      data-radix-aspect-ratio-wrapper=""
      style="position: relative; width: 100%; padding-bottom: 75%;"
    >
      <div
        style="position: absolute; top: 0px; right: 0px; bottom: 0px; left: 0px;"
      >
        <img
          alt="A mock title"
          class="rounded"
          src="/images/mock.jpg"
        />
      </div>
    </div>
    <div
      class="flex flex-col px-1"
    >
      <div
        class="flex w-full justify-between"
      >
        <div
          class="flex flex-col"
        >
          <h2
            class="text-base sm:text-lg font-semibold"
          >
            A mock title
          </h2>
          <p
            class="text-sm sm:text-base text-gray-400"
          >
            A mock subtitle
          </p>
        </div>
        <span
          class="text-gray-400 text-nowrap w-fit text-sm sm:text-base"
        >
          Mock author
        </span>
      </div>
      <div
        class="w-full text-sm sm:text-base mt-2 text-lines-limit-3"
      >
        A mock content with a lot of text to fill the space
      </div>
      <div
        class="flex justify-end"
      >
        <button
          class="text-gray-500 hover:text-gray-300 text-right text-sm sm:text-base"
        >
          Read more
        </button>
      </div>
    </div>
    <hr
      class="border-gray-800 mx-[-8px]"
    />
    <div
      class="px-1"
    >
      <button
        class="text-gray-500 hover:text-gray-300"
      >
        View 
        2
         comment
        s
      </button>
    </div>
  </div>
</div>
`;
